tenant-isolation-config-service-impl/src/main/java/org/hypertrace/tenantisolation/config/service/TenantIsolationConfigServiceImpl.java

```diff
@@ -0,0 +1,141 @@
+package org.hypertrace.tenantisolation.config.service;
+
+import com.google.inject.Inject;
+import com.typesafe.config.Config;
+import io.grpc.stub.StreamObserver;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.hypertrace.tenantisolation.config.service.store.TenantIsolationConfigStore;
+import org.hypertrace.tenantisolation.config.service.store.TenantIsolationConfigStoreProvider;
+import org.hypertrace.tenantisolation.config.service.store.TenantIsolationGroupConfigDTO;
+import org.hypertrace.tenantisolation.config.service.v1.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TenantIsolationConfigServiceImpl
+    extends TenantIsolationConfigServiceGrpc.TenantIsolationConfigServiceImplBase {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(TenantIsolationConfigServiceImpl.class);
+
+  private static final String GENERIC_CONFIG_SERVICE = "generic.config.service";
+
+  private TenantIsolationConfigServiceRequestValidator validator =
+      new TenantIsolationConfigServiceRequestValidator();
+
+  private TenantIsolationConfigStore tenantIsolationConfigStore;
+
+  @Inject
+  public TenantIsolationConfigServiceImpl(Config config) {
+    Config cfg = config.getConfig(GENERIC_CONFIG_SERVICE);
+    this.tenantIsolationConfigStore = TenantIsolationConfigStoreProvider.getDocumentStore(cfg);
+  }
+
+  public void getAllTenantIsolationGroupConfigs(
+      GetAllTenantIsolationGroupConfigsRequest request,
+      StreamObserver<GetAllTenantIsolationGroupConfigsResponse> responseObserver) {
+    try {
+      validator.validateOrThrow(request);
+      List<TenantIsolationGroupConfigDTO> configDTOs =
+          tenantIsolationConfigStore.getAllGroupConfigs();
+      List<TenantIsolationGroupConfig> configs =
+          configDTOs.stream()
+              .map(
+                  tenantIsolationGroupConfigDTO ->
+                      TenantIsolationGroupConfig.newBuilder()
+                          .setGroupName(tenantIsolationGroupConfigDTO.getGroupName())
+                          .setWeight(tenantIsolationGroupConfigDTO.getWeight())
+                          .addAllMembers(tenantIsolationGroupConfigDTO.getMembers())
+                          .build())
+              .collect(Collectors.toList());
+
+      responseObserver.onNext(
+          GetAllTenantIsolationGroupConfigsResponse.newBuilder().addAllConfigs(configs).build());
+      responseObserver.onCompleted();
+    } catch (Exception e) {
+      logger.error("getAllTenantIsolationGroupConfigs failed ", e);
+      responseObserver.onError(e);
+    }
+  }
+
+  public void createTenantIsolationGroupConfig(
+      CreateTenantIsolationGroupConfigRequest request,
+      StreamObserver<CreateTenantIsolationGroupConfigResponse> responseObserver) {
+    try {
+      validator.validateOrThrow(request);
+      TenantIsolationGroupConfigDTO upsertedConfigDTO =
+          tenantIsolationConfigStore.upsert(
```

Comment:
What's the purpose of this DTO? Seems like complexity for no benefit. Given we have equivalent compatibility requirements on both API and storage, generally we use the same item. Even if they ever were to diverge in the future, we can still do that just by diverging the implementation.